tests/partitions_test.py

```diff
@@ -148,6 +150,56 @@ def test_isolate_and_reconnect_primary(network, args, **kwargs):
         assert status == TxStatus.Invalid, r
 
 
+@reqs.description("New joiner helps liveness")
+def test_new_joiner_helps_liveness(network, args):
+    primary, backups = network.find_nodes()
+
+    # Issue some transactions, so there is a ledger history that a new node must receive
+    network.txs.issue(network, number_txs=10)
+
+    # Remove a node, leaving the network frail
+    network.retire_node(primary, backups[-1])
+    backups[-1].stop()
+
+    primary, backups = network.find_nodes()
+
+    # Add a new node, but partition them before trusting them
+    new_node = network.create_node("local://localhost")
+    network.join_node(new_node, args.package, args, from_snapshot=False)
+    new_joiner_partition = [new_node]
+    new_joiner_rules = network.partitioner.partition(
```

Comment:
Could you use the context manager variant for this? I believe it would make the test slightly clearer since there are two partitions going on and would also automatically delete the partition if an exception occurs.

Reply:
We could use a context manager for one of these, but not both (easily), because they're opened and dropped in the same order, not nested. ie `create_partition(A); create_partition(B); drop_partition(B); drop_partition(A);` can be converted to context managers, but we need the exit stack thing for this `create_partition(A); create_partition(B); drop_partition(A); drop_partition(B);` pattern. I'll see if that works.

Reply:
Turns out this _is_ context stack-able, I didn't know you could still get the return values to manually drop. Cool.